ttnn/cpp/ttnn/operations/pool/upsample/device/kernels/dataflow/writer_upsample_multi_core_sharded.cpp

```diff
@@ -16,50 +33,39 @@ void kernel_main() {
     constexpr uint32_t in_cb_id = get_compile_time_arg_val(0);
     constexpr uint32_t out_cb_id = get_compile_time_arg_val(1);
     constexpr uint32_t is_reader = get_compile_time_arg_val(2);
+    constexpr uint32_t config_cb_id = get_compile_time_arg_val(3);
+
+    uint32_t reader_nsticks_per_core = (in_nsticks_per_core + is_reader) / 2;
+    uint32_t writer_nsticks_per_core = in_nsticks_per_core / 2;
```

Comment:
I have removed this code and moved output sticks logic here and am deriving output sticks from reader_nsticks_per_core so that we handle even and odd case correctly.